cli/githubutil.py

```diff
@@ -140,3 +142,99 @@ def release_note_blocks_cli(
         repository_branch=repository_branch,
         commit_range=commit_range
     ).release_note_blocks()
+
+
+def cleanup_draft_releases(
+    github_cfg_name: str,
+    github_repository_owner: str,
+    github_repository_name: str,
+    greatest_release_version: CliHint(
+        help=(
+            'Greatest version (according to Semver) up to which (exclusive) '
+            'to consider draft releases.'
+        ),
+    ),
+    dry_run: bool=True,
+):
+    '''Remove outdated draft releases from a GitHub repository.
+
+    Scans all releases on GitHub and removes outdated draft releases. A draft release is considered
+    outdated iff:
+        1: it is preceding the greatest (according to Semver) release AND
+            2a: it is NOT a hotfix draft release AND
+            3a: there are no hotfix draft releases with the same major and minor version
+            OR
+            2b: it is a hotfix draft release AND
+            3b: there is a hotfix draft release with the same major and minor version succeeding it
+
+    This is achieved by partitioning the releases according to their major and minor version
+    combined. Partitions are then checked:
+        - if there is only a single release in a partition it is either a hotfix release (keep)
+        or it is not (delete)
+        - if there are multiple releases in a partition, keep only the greatest (according
+        to semver)
+    '''
+    github_cfg = ctx().cfg_factory().github(github_cfg_name)
+    githubrepobranch = GitHubRepoBranch(
+        github_config=github_cfg,
+        repo_owner=github_repository_owner,
+        repo_name=github_repository_name,
+        branch=None,
+    )
+
+    github_helper = GitHubRepositoryHelper.from_githubrepobranch(
+        githubrepobranch=githubrepobranch,
+    )
+
+    greatest_semver = semver.parse_version_info(greatest_release_version)
+
+    def _is_autogenerated_draft_release(release):
+        try:
+            release_semver = semver.parse_version_info(release.name)
+        except Exception:
+            return False
+        if release_semver.prerelease != 'draft':
+            return False
+        return True
+
+    autogenerated_draft_releases = [
+        release for release in github_helper.repository.releases()
+        if release.draft and _is_autogenerated_draft_release(release)
```

Comment:
I'm confused: in `is_autoge..._draft_release`, you check whether release_semver.prerelease == 'draft'. here, it is checked whether release.draft is true-ish. is this redundant? can there be cases where the two checks bear different results? if so, would this not indicate an error condition? either in the cfg, or in our coding?